policy/modules/contrib/pcp.te

```diff
@@ -111,6 +111,7 @@ allow pcp_pmcd_t self:capability { dac_read_search dac_override ipc_owner net_ad
 allow pcp_pmcd_t self:process { setsched };
 allow pcp_pmcd_t self:unix_dgram_socket create_socket_perms;
 allow pcp_pmcd_t self:cap_userns sys_ptrace;
+allow pcp_pmcd_t self:perf_event open;
```

Comment:
This will definitely need more than just `open`. The original test run was done in enforcing mode, so it only produced one AVC. After running it in permissive mode I got:
```
type=PROCTITLE msg=audit(1606909561.538:403): proctitle=2F7661722F6C69622F7063702F706D6461732F706572666576656E742F706D6461706572666576656E74002D6400313237
type=SYSCALL msg=audit(1606909561.538:403): arch=c000003e syscall=298 success=no exit=-2 a0=56085c9c93b8 a1=ffffffff a2=0 a3=ffffffff items=0 ppid=40302 pid=40787 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="pmdaperfevent" exe="/usr/libexec/pcp/pmdas/perfevent/pmdaperfevent" subj=system_u:system_r:pcp_pmcd_t:s0 key=(null)
type=AVC msg=audit(1606909561.538:403): avc:  denied  { kernel } for  pid=40787 comm="pmdaperfevent" scontext=system_u:system_r:pcp_pmcd_t:s0 tcontext=system_u:system_r:pcp_pmcd_t:s0 tclass=perf_event permissive=1
type=AVC msg=audit(1606909561.538:403): avc:  denied  { open } for  pid=40787 comm="pmdaperfevent" scontext=system_u:system_r:pcp_pmcd_t:s0 tcontext=system_u:system_r:pcp_pmcd_t:s0 tclass=perf_event permissive=1
```
So at least `kernel` is needed as well. However, the syscall above failed with `-ENONENT`, so it is very likely that the program would also need `read` and/or `write` in a different code path.

Reply:
So we can probably use the same `manage_perf_event_perms' pattern as for users.

Reply:
Yeah, I suppose that would be OK... Unless we want to be strict about the principle of least privileges, but I guess it's not really worth it here (the difference ATM would be just `cpu` and possibly `write` or `read`).

Reply:
So I ran the test on F33 and there it wanted the full set:
```
----
time->Wed Dec  2 09:06:52 2020
type=AVC msg=audit(1606918012.254:283): avc:  denied  { open } for  pid=18335 comm="pmdaperfevent" scontext=system_u:system_r:unconfined_service_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=perf_event permissive=1
----
time->Wed Dec  2 09:06:52 2020
type=AVC msg=audit(1606918012.254:284): avc:  denied  { kernel } for  pid=18335 comm="pmdaperfevent" scontext=system_u:system_r:unconfined_service_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=perf_event permissive=1
----
time->Wed Dec  2 09:06:52 2020
type=AVC msg=audit(1606918012.254:285): avc:  denied  { cpu } for  pid=18335 comm="pmdaperfevent" scontext=system_u:system_r:unconfined_service_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=perf_event permissive=1
----
time->Wed Dec  2 09:06:52 2020
type=AVC msg=audit(1606918012.255:286): avc:  denied  { read } for  pid=18335 comm="pmdaperfevent" scontext=system_u:system_r:unconfined_service_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=perf_event permissive=1
----
time->Wed Dec  2 09:06:52 2020
type=AVC msg=audit(1606918012.356:287): avc:  denied  { write } for  pid=18335 comm="pmdaperfevent" scontext=system_u:system_r:unconfined_service_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=perf_event permissive=1
[...]
```
(For some reason pcp ran as `unconfined_service_t` there... not sure why.)

Reply:
And in 1minutetip f33 I also got some denials for `pmdakvm`, which also wanted `tracepoint`:
```
----
type=AVC msg=audit(02.12.2020 09:55:27.383:679) : avc:  denied  { open } for  pid=23019 comm=pmdakvm scontext=system_u:system_r:unconfined_service_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=perf_event permissive=1 
----
type=AVC msg=audit(02.12.2020 09:55:27.383:680) : avc:  denied  { kernel } for  pid=23019 comm=pmdakvm scontext=system_u:system_r:unconfined_service_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=perf_event permissive=1 
----
type=AVC msg=audit(02.12.2020 09:55:27.384:681) : avc:  denied  { tracepoint } for  pid=23019 comm=pmdakvm scontext=system_u:system_r:unconfined_service_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=perf_event permissive=1 
----
type=AVC msg=audit(02.12.2020 09:55:27.384:682) : avc:  denied  { cpu } for  pid=23019 comm=pmdakvm scontext=system_u:system_r:unconfined_service_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=perf_event permissive=1 
----
type=AVC msg=audit(02.12.2020 09:55:27.384:683) : avc:  denied  { write } for  pid=23019 comm=pmdakvm scontext=system_u:system_r:unconfined_service_t:s0 tcontext=system_u:system_r:unconfined_service_t:s0 tclass=perf_event permissive=1 
```